docs: Archive Tendermint Core ADRs/RFCs (#393)
cometbft/cometbft: docs/architecture/README.md

# Architecture Decision Records (ADR)

This is a location to record all high-level architecture decisions in the
CometBFT project.

You can read more about the ADR concept in this
[blog post](https://product.reverb.com/documenting-architecture-decisions-the-reverb-way-a3563bb24bd0#.78xhdix6t).

An ADR should, with a strong focus on the impact on _users_ of the system,
provide:

- Context on the relevant goals and the current state
- Proposed changes to achieve the goals
- Summary of pros and cons
- References
- Changelog

> Note the distinction between an ADR and a spec. The ADR provides the context,

Reviewer: ```suggestion
Note the distinction between an ADR and a spec. An ADR provides the context,
```

intuition, reasoning, and justification for a change in architecture, or for the

> architecture of something new. The spec is much more compressed and streamlined

Reviewer: ```suggestion
architecture of something new. A spec is more compressed and streamlined
```
